### PR TITLE
EM-3473 - EM-Anno api: Add functional test cases for RectangleResource

### PR DESCRIPTION
**Pull request by rajkatla-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/CommentScenarios.java [hunk 1]
```diff
@@ -247,8 +247,17 @@ public void shouldReturn200WhenDeleteCommentById() {
...
-        deleteCommentById(id).statusCode(200);
+        deletedResponse.statusCode(200);
+    }
+
+    @Test
+    public void shouldReturn500WhenDeleteCommentById() {
+        final String commentId = UUID.randomUUID().toString();
+        final ValidatableResponse deletedResponse = deleteCommentById(commentId);
+
+        deletedResponse.statusCode(500);
```
Comment (benouaer) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/CommentScenarios.java:260]: why is it supposed to return 500?

Comment (rajkatla-hmcts) [reply src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/CommentScenarios.java:260]: at the moment api returning 500 when given id is not found and raised a bug for this issue. API should return 404 when id not found. For now added this test case and when the bug is fixed this test case will fail, so we need to update this test case to make it to pass.

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+    @Test
+    public void shouldReturn500WhenCreateNewRectangle() {
+        final String annotationId = UUID.randomUUID().toString();
+        final String rectangleId = UUID.randomUUID().toString();
+        final JSONObject rectanglePayload = createRectanglePayload(annotationId, rectangleId);
+        request
+                .body(rectanglePayload.toString())
+                .post("/api/rectangles")
+                .then()
+                .statusCode(500)
```
Comment (benouaer) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:118]: why?

Comment (rajkatla-hmcts) [reply src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:118]: its same as above issue, raised a bug for this issue as well.

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+    @Test
+    public void shouldReturn500WhenUpdateRectangle() {
+        final String rectangleId = UUID.randomUUID().toString();
+        final String annotationId = UUID.randomUUID().toString();
+        final JSONObject rectangle = createRectanglePayload(annotationId, rectangleId);
+        request
+                .body(rectangle.toString())
+                .put("/api/rectangles")
+                .then()
+                .statusCode(500)
```
Comment (benouaer) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:219]: again, it's not clear why

Comment (rajkatla-hmcts) [reply src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:219]: its same as above issue, raised a bug for this issue as well.

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+
+        deletedResponse.statusCode(200);
+    }
+
+    @Test
+    public void shouldReturn500WhenDeleteRectangleById() {
+        final String rectangleId = UUID.randomUUID().toString();
+        final ValidatableResponse deletedResponse = deleteRectangleById(rectangleId);
+
+        deletedResponse.statusCode(500);
```
Comment (benouaer) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:272]: can't see why

File: src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+        final String id = rectangle.getString("id");
+        deleteRectangleById(id).statusCode(200);
+        rectangle.put("x", 3f);
+        rectangle.put("y", 4f);
+
+        request
+                .body(rectangle.toString())
+                .put("/api/rectangles")
+                .then()
+                .statusCode(200)
```
Comment (benouaer) [inline src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:300]: shouldn't this return 404 since the resource doesn't exist?

Comment (rajkatla-hmcts) [reply src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java:300]: At the moment update is creating new resource when it does not find the entity with given id and updating if it found.

**Review by benouaer — APPROVED** [review]

**Pull request merged**